CONTRIBUTING.md

```diff
@@ -0,0 +1,270 @@
+working on the same feature or bug fixe (yes, it has happened), we recommend
+strongly that you raise a github issue **before** you start working on a PR.
+
+If you are a new contributor you will not be able to assign the issue to
+yourself so please raise an issue and add a comment stating that you intend to
+work on the issue yourself. This gives the team visibility of the work you
+plan to do.
+
+The other advantage to raising the issue at the outset is that our process requires an
+issue to be associated with every PR (see [patch format](#patch-format)).
+
+### Normal PR workflow
+
+Github has a basic introduction to the PR process
+[here](https://help.github.com/articles/using-pull-requests/).
+
+When submitting your PR, treat the PR message the same
+you would a patch message, including pre-fixing the title with a subsystem
+name. Github by default seems to copy the message from your first patch, which
+many times is appropriate, but please ensure your message is accurate and
+complete for the whole PR, as it ends up in the git log as the merge
+message.
+
+Your PR may get some feedback and comments, and require some rework.
+The recommended procedure for reworking is to rework your branch to a new clean
+state and "force push" it to your github branch. Github understands this action, and
+does sensible things in the online comment history. Do not pile patches on
+patches to rework your branch. Any relevant information from the github
+comments section should be re-worked into your patch set, as the ultimate place
+where your patches are documented is in the git log, and not in the github
+comments section.
+
+For more information on github "force push" workflows see
+[here](http://blog.adamspiers.org/2015/03/24/why-and-how-to-correctly-amend-github-pull-requests/).
+
+It is perfectly fine for your PR to contain more than one patch - use
+as many patches as you need to implement the Request (see the previously
+mentioned "small patch" thoughts). Each PR should only cover one
+topic - if you mix up different items in your patches or PR then you
+will most likely be asked to rework them.
+
+### Re-vendor PRs
+
+If you are raising a PR to update the vendored copy of one or more golang
+packages, there are two critical pieces of information you need to add to the
+commit body:
+
+- A brief explanation as to why this re-vendor is required.
+
+  For example, it might be that an important bug fix or new feature is
+  required. It might even be that a particular commit is needed. If so, please
+  state that.
+
+- The range of commits being added for these third-party packages.
+
+  It is possible that re-vendoring a particular package will also result in updates to
+  other dependent packages. However, it is important to include the commit range
+  (even if it is a big) for the primary package(s) the re-vendor PR has been
+  raised for.
+
+  These details allow for easier problem-determination in the future should
```

Comment:
s/problem-determination/problem determination/

Reply:
Yep. It seems that hyphens are becoming deprecated everywhere (sob), so removing...